src/main/java/listeners/OnlinerTestListener.java

```diff
@@ -39,19 +42,24 @@ public void onTestFailedWithTimeout(ITestResult result) {
     @Override
     public void onStart(ITestContext context) {
         logger.info("The test class run started on " + context.getStartDate());
-        allureEnvironmentWriter(
-                ImmutableMap.<String, String>builder()
-                        .put("Browser", "Chrome")
-                        .put("Browser.Version", "96.0.4664.93 (Official build), (64 bit)")
-                        .build(), System.getProperty("user.dir")
-                        + "/target/allure-results/");
+        context.getCurrentXmlTest().getSuite().setParallel(XmlSuite.ParallelMode.METHODS);
+        context.getCurrentXmlTest().getSuite().setThreadCount(ThreadCountConfig.getThreads());
     }
 
     @Override
     public void onFinish(ITestContext context) {
         logger.info("The test class run finished on ... " + context.getEndDate());
     }
 
+    public static void attachEnvironmentInfo(WebDriver driver ) {
```

Comment:
Why do you pass driver as an argument? can't you access driver in any framework layer without that?

Reply:
Yes, I can access it. The thing is that when tests run, 2 drivers open , one in the test and one in attachEnvironmentInfo().